src/components/NewCommentForm.tsx

```diff
@@ -1,75 +1,109 @@
 import classNames from 'classnames';
 import React, { useState } from 'react';
-import { CommentData } from '../types/Comment';
-
-type Props = {
-  onSubmit: (data: CommentData) => Promise<void>;
-};
-
-export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
-  const [submitting, setSubmitting] = useState(false);
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { newFormSlice } from '../features/newCommentForm';
+import { addComment } from '../features/comments';
 
+export const NewCommentForm = () => {
+  const dispatch = useAppDispatch();
+  const { name, email, body } = useAppSelector(state => state.newForm);
   const [errors, setErrors] = useState({
     name: false,
     email: false,
     body: false,
   });
+  const [submitting, setSubmitting] = useState(false);
+  const { item: selectedPost } = useAppSelector(state => state.selectedPost);
 
-  const [{ name, email, body }, setValues] = useState({
-    name: '',
-    email: '',
-    body: '',
-  });
-
-  const clearForm = () => {
-    setValues({
-      name: '',
-      email: '',
-      body: '',
-    });
-
-    setErrors({
-      name: false,
-      email: false,
-      body: false,
-    });
+  const handleClearTwoFiledForm = () => {
+    dispatch(newFormSlice.actions.clearTwoFiledForm());
   };
 
-  const handleChange = (
-    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
-  ) => {
-    const { name: field, value } = event.target;
+  const handleSubmiting = (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
There is a typo in the function name `handleSubmiting`. It should be `handleSubmitting` to maintain consistency and clarity in naming conventions.